### ISSUE TITLE
extraCurrency transfer fail (on private testnet)

### ISSUE THREAD

**Issue by EmelyanenkoK** [ton-blockchain/ton#261]
I'm not actually sure is it a real issue or I incorrectly deployed private testnet. I'm trying to test transfers of additional currencies to support this feature in gram-wallet.
As for now, I managed to deploy private testnet, have added to the state of main-wallet account additional currency with id 0, have generated valid zero-state. Validation starts, then I tried to send 1 unit of $0 currency and collator immediately stopped generate valid blocks with the following error (this error occurs after block generation not in the process of validation):
```
[t 5][1582786155.053308010][collator.cpp:3622][!collate(-1,8000000000000000):51]	incorrect value flow in new block : (value-flow from_prev_blk:(7699991770750000000ng+100000.$0+666666666666.$239+1000000000000.$-17) to_next_blk:(7699991772450000000ng+1.$0) imported:230000000ng exported:230000000ng fees_collected:1720000000ng fees_imported:0ng recovered:1720000000ng created:1700000000ng minted:0ng)
```

Looks like value flow is calculated incorrectly. If my setup is correct, the problem may be in a few places:
1. get_root_extra which relies on summation of children for Augmented Hashmaps may be incorrect itself (https://github.com/ton-blockchain/ton/blob/f67f5d879ba0a395367a703dcd75f3b87c6ad535/validator/impl/collator.cpp#L3520), although I don't see a problem in the code.
2. account -1:0000...000 may be somehow excluded from the calculation if it is not the only leaf in dictionary

**Comment by ton-blockchain** [comment]
Well, zero is not a good `extra-currency-id`. But the bug could exist for non-zero currency ids as well. We'll try to reproduce it locally using `test-ton-collator`.

**Comment by ton-blockchain** [comment]
We managed to reproduce this bug. The situation is even worse than it seems. We'll provide a bugfix and more details later.

**Comment by ton-blockchain** [comment]
We think that this bug has been fixed in the latest commit dd4ac0f44083316a005c01e48fb81f3a64d5d79b. Please check.

Among other things, this commit contains an updated `wallet.fif` that accepts a new command line option `-x` to indicate extra currencies, e.g. `-x '11223344*$77+4444*$-5'`.

**Comment by EmelyanenkoK** [comment]
Solved in dd4ac0f
Thanks!
P.S. Not sure why $0 is bad currency id, though. Anyway successfully transferred units of `$0` using wallet.fif
